src/main/java/org/mqttbee/mqtt/handler/publish/MqttIncomingAckFlow.java

```diff
@@ -36,6 +36,10 @@
  */
 public class MqttIncomingAckFlow implements Subscription, Runnable {
 
+    private static final int STATE_NO_NEW_REQUESTS = 0;
```

Comment:
use enum for states

can we reuse states from MqttIncomingPublishFlow

Reply:
Enum is not used because we are dealing with atomic variables

Reply:
Why can't we use AtomicReference?

Reply:
We should not use AtomicReference if not necessary, as it does more magic as just setting the address of the object as an AtomicInteger or AtomicLong (GC house-keeping or something like that).
I did a quick and dirty performance test, which works because the JIT can not optimize Atomics.
```
enum Test {
    V1, V2, V3
}

    public static void main(String[] args) {
        final AtomicReference<Test> reference = new AtomicReference<>();
        final AtomicInteger integer = new AtomicInteger();
        final AtomicLong atomicLong = new AtomicLong();
        final long before = System.currentTimeMillis();
        for (int i = 0; i < 1000_000_000; i++) {
            reference.set(Test.V1);
            reference.get();
//            integer.set(i);
//            integer.get();
//            atomicLong.set(i);
//            atomicLong.get();
        }
        System.out.println(System.currentTimeMillis() - before);
    }
```
Results (for 1000_000_000 iterations):
- AtomicInteger: 5,4s
- AtomicLong: 5,6s
- AtomicReference: 7,1s (this is approx. 30% slower than AtomicInteger)